crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,437 @@
+//! Components used to control the layout of [`UiNode`] entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+/// Controls which layout algorithm is used to position this UI node
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum LayoutStrategy {
+    /// Use the absolute position and size specified
+    None,
+    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm
+    ///
+    /// As implemented by [`taffy`]: some bugs or limitations may exist; please file an issue!\
+    #[default]
+    Flex,
+}
+
+/// The strategy used to position this node
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum PositionType {
+    /// Relative to all other nodes with the [`PositionType::Relative`] value
+    #[default]
+    Relative,
+    /// Positioned as if it was the only child of its parent
+    ///
+    /// As usual, the `Style.position` field of this node is specified relative to its parent node
+    Absolute,
+}
+
+/// The offset of a UI node from its base position
+///
+/// Layout is performed according to the [`LayoutStrategy`], and then this value is added at the end.
+/// When this is [`LayoutStrategy::None`], this value will represent the absolute position of the UI node.
+/// To check the final position of a UI element, read its [`Transform](bevy_transform::Transform) component.
+#[derive(
+    Component,
+    Deref,
+    DerefMut,
+    Copy,
+    Clone,
+    PartialEq,
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct Offset(pub UiRect<Val>);
+
+/// Controls the size of UI nodes
+///
+/// Layout is performed according to the [`LayoutStrategy`]
+/// To check the actual size of a UI element, read its [`Transform](bevy_transform::Transform) component
+#[derive(Component, Copy, Clone, PartialEq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct SizeConstraints {
+    /// The minimum extent, which cannot be violated by the layouting algorithm
+    pub min: Size<Val>,
+    /// The suggested extent, which will be used if other constraints can be comfortably satisfied
+    pub suggested: Size<Val>,
```

Comment:
I think the documentation needs to be more specific here: Which other constraints need to be satisfied? Does this value take precedence over min and max or the other way around?
Since we have two axis, what happens if one is satisfied and the other not?